src/Tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema/SchemaGenerator.cs

```diff
@@ -191,6 +202,16 @@ private static void AddCollectionRuleTriggerSchema<TOptions>(GenerationContext c
 
             settingsProperty.Reference = context.AddTypeIfNotExist<TOptions>();
 
+            foreach (var propName in typeof(CollectionRuleDefaultsOptions).GetProperties().Select(x => x.Name))
+            {
+                settingsProperty.Reference.RequiredProperties.Remove(propName);
+            }
+
+            if (settingsProperty.Reference.RequiredProperties.Count > 0)
```

Comment:
This is almost correct; you have to check that the required properties don't have default values and there is no corresponding default property.